### PR TITLE
gRPC server implementation

### PR DESCRIPTION
**Pull request by alrevuelta**
Implements a gRPC server that exposes a service where all crawler clients can report the peers they discover.

TODO:
- [ ] Remove hardcoded parameters in favor of cli flags
- [ ] Rethink the metadata parameters to be sent
- [ ] Place the gRPC call in the correct place
- [ ] TLS certificates or some other kind of authentication mechanism
- [ ] Implement the server logic, store peers and update them
- [ ] Export metrics to prometheus, reusing the already existing code in the armiarma-client. Note that some modifications may be needed in order to be reusable.
- [ ] Write unit tests
- [ ] Integration tests with different machines reporting to a remote endpoint

### CODE REVIEW

File: src/proto/aggregator.proto [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  string crawler_id = 1;
+  string peer_id = 2;
+  string node_id = 3;
+  string client_type = 4;
+  string pub_key = 5;
+  string address = 6;
+  string country = 7;
+  string city = 8;
+  string latency = 9;
+  // TODO: inbound/outbound
```
Comment (alrevuelta) [inline src/proto/aggregator.proto:24]: @Cortze Do we have access to the inbound/outbound field?

Comment (Cortze) [reply src/proto/aggregator.proto:24]: Yep, we keep track of the origin and status of each of the connections, even the error reported when the connection failed

File: src/metrics/gossip_metrics.go [hunk 2]
```diff
@@ -26,12 +29,26 @@ type GossipMetrics struct {
 	MessageDatabase *database.MessageDatabase
 	StartTime       int64 // milliseconds
 	MsgNotChannels  map[string](chan bool)
+	AggregatorClient pb.AggregatorClient
 }
 
 func NewGossipMetrics() GossipMetrics {
+	// TODO: Just a proof of concept. This has to be moved
```
Comment (Cortze) [inline src/metrics/gossip_metrics.go:36]: This hardcoded implementation of the host dialing is solved in the generated PR

File: src/metrics/gossip_metrics.go [hunk 3]
```diff
@@ -414,6 +431,28 @@ func (c *GossipMetrics) AddMetadataEvent(peerId peer.ID, success bool) {
...
+			fmt.Println("peerMetrics.ClientType", peerMetrics.ClientType)
+			fmt.Println("peerMetrics.Pubkey", peerMetrics.Pubkey)
+			fmt.Println("peerMetrics.Addrs", peerMetrics.Addrs)
+			fmt.Println("peerMetrics.Ip", peerMetrics.Ip)
+			fmt.Println("peerMetrics.Country", peerMetrics.Country)
+
+			ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+		  defer cancel()
+
+			r, err := c.AggregatorClient.NewPeerMetadata(ctx, &pb.NewPeerMetadataRequest{
```
Comment (Cortze) [inline src/metrics/gossip_metrics.go:445]: Added into the /serverexport submodule in the PR

File: src/proto/aggregator.proto [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+message NewPeerMetadataReply {
+  string dummy = 1;
```
Comment (Cortze) [inline src/proto/aggregator.proto:28]: Perhaps changing the name to ACK or something similar?

**Comment by Cortze** [comment]
Closing PR due to integration of PostgreSQL as the main database in #35. All the information of the DB endpoint will be now accessible from the outside, so there is no need for a Crawler - Server communication.

**Pull request closed without merging**